roles/customize_home/tasks/clone_specified_repo.yml

```diff
@@ -0,0 +1,39 @@
+- name: Ensure temporary home directory does not exist
+  file:
+    path: "{{ ansible_env.HOME }}/tmp_custom_home/"
+    state: absent
+
+- name: Clone specified repo into temporary custom home directory
+  git:
+    repo: "{{ customize_home_git_repo }}"
+    dest: "{{ ansible_env.HOME }}/tmp_custom_home/"
+
+- name: Remove .git directory from temporary home directory after cloning
+  file:
+    path: "{{ ansible_env.HOME }}/tmp_custom_home/.git"
+    state: absent
+
+- name: Remove .gitignore from temporary home directory after cloning
+  file:
+    path: "{{ ansible_env.HOME }}/tmp_custom_home/.gitignore"
+    state: absent
+
+# Using command module because the copy module won't match hidden (dot) files and using copy with_fileglob isn't working as expected
+# see https://github.com/ansible/ansible/issues/17136 for more info on with_fileglob
+- name: Copy temporary home directory to actual home directory
+  command: "cp -r {{ ansible_env.HOME }}/tmp_custom_home/. {{ ansible_env.HOME }}/"
+
+- name: Remove temporary home directory
+  file:
+    path: "{{ ansible_env.HOME }}/tmp_custom_home/"
+    state: absent
+
+- name: Check custom bootstrap script exists
+  stat:
+    path: "{{ ansible_env.HOME }}/{{ custom_bootstrap_script }}"
+  register: bootstrap_script_path
+
+# The script is already on the machine, so using 'command' instead of 'script', which copies the script from the Ansible controller
+- name: Run custom bootstrap script
```

Comment:
If I have a custom script, I expect that script to be run and control everything that happens.

Reply:
@ehelms can you elaborate on this? The script is run on the machine after being copied over.

Reply:
It is related to my previous comment around order of operations. I tend to look at it as, if I provide a script, the script should be all that happens and thus I have complete control over everything. Whereas if no script is supplied, I expect the simple case of copying dotfiles.

Reply:
@ehelms hopefully I addressed this above, but if you still have concerns maybe we can poll some other devs for input.